### PR TITLE
Update the CLion instructions to use direnv

### PR DESCRIPTION
**Pull request by knopers8**
@ktf
@Barthelemy Could you try it out when you are back?

### CODE REVIEW

File: advanced/ides.markdown [hunk 1]
```diff
@@ -9,18 +9,30 @@ CLion
...
-4. Access the cmake configuration (see number 1 on figure below)
-5. Edit the environment variables (see number 2 on figure below)
-6. Paste the clipboard content using the button labelled 3 in the figure
+1. Install [`direnv`](https://direnv.net/docs/installation.html).
+2. Build with aliBuild as you would do normally.
+3. Go to the directory where you run CLion.
+4. Create a new file ".envrc" which loads all the build environments of the projects that you will run on CLion. Use ~/alice/sw/BUILD/O2-latest/O2/.envrc as a starting point. For example:
+```
+# Source the build environment which was used for this package
+WORK_DIR=/home/pkonopka/alice/sw
```
Comment (ktf) [inline advanced/ides.markdown]: I would put `/<workdir>` rather than your `$HOME`...

Comment (knopers8) [reply advanced/ides.markdown]: indeed

**Comment by Barthelemy** [comment]
It seems very promising but unfortunately I did not manage to make it work on my mac. Common, InfoLogger, arrow and fairmq need to be set manually (i.e. -Darrow_DIR=...). Once I set them, I get an error due to DebugGui : `Target "QualityControl" links to target "AliceO2::DebugGUI" but the target  was not found.`

I am now trying on CC7.

**Comment by Barthelemy** [comment]
On CC7 I have the same problem but with Configuration, ROOT, Monitoring...

I tried with the defaults `o2` after building QualityControl.

**Comment by Barthelemy** [comment]
Victory !

As I already had a project, it was not enough to rerun CMake within clion, I had to trash the project. Factually, I had to remove `.idea` from the source directory, e.g. `QualityControl/.idea`.
This could probably be added. 

Another catch on mac is that you have to find where the binary actually is. If one uses the "Jetbrain toolbox" then the path is very circumvoluted : `~/Library/Application\ Support/JetBrains/Toolbox/apps/CLion/ch-0/201.5616.31/CLion\ 2020.1\ EAP.app/Contents/MacOS/clion` (and there are bits to update upon new versions). 
In case clion is installed in `/Application` it would be slightly easier. Maybe a note can also be added about that (I can't do it as I am not allowed to edit hte PR). 

Thank you very much, it is great !

Kostas is looking into remote development with Clion and that would be a great combo.

**Comment by ktf** [comment]
Great. Can we merge this then?

**Comment by knopers8** [comment]
I would propose to merge it and Barth would extend it himself.

It would be great if @kostorr could leave some intstructions if he succeeds to configure clion for remote development.

**Comment by kostorr** [comment]
> I would propose to merge it and Barth would extend it himself.
> 
> It would be great if @kostorr could leave some intstructions if he succeeds to configure clion for remote development.

Sure, I will make sure to add anything I find.

**Pull request merged**